uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/staff-leave-report.component.js

```diff
@@ -121,18 +121,20 @@ define([
     }
 
     /**
-     * Calls the load function of the given data, and puts the section
-     * in and out of loading mode
+     * Calls the load function of the given data, indexes the loaded data,
+     * and puts the section in and out of loading mode
      *
      * @param  {Object} section
      * @return {Promise}
      */
     function callSectionLoadFn (section) {
       section.loading = true;
 
-      return section.loadFn().then(function () {
-        section.loading = false;
-      });
+      return section.loadFn()
```

Comment:
If you refactored this anyway, may I ask you please to find a better name for the `loadFn` func?